src/client/content_cao.cpp

```diff
@@ -1165,14 +1165,16 @@ void GenericCAO::step(float dtime, ClientEnvironment *env)
 			m_step_distance_counter = 0.0f;
 			if (!m_is_local_player && m_prop.makes_footstep_sound) {
 				const NodeDefManager *ndef = m_client->ndef();
-				v3s16 p = floatToInt(getPosition() +
-					v3f(0.0f, (m_prop.collisionbox.MinEdge.Y - 0.5f) * BS, 0.0f), BS);
-				MapNode n = m_env->getMap().getNode(p);
+				v3f foot_pos = getPosition() * (1.0f/BS)
```

Comment:
use `floatToInt` here, otherwise the rounding is incorrect

Reply:
`floatToInt(p/BS, 1.0f)`:
`out.x = (p.X/BS + (p.X/BS > 0 ? 1.0f / 2 : -1.0f / 2)) / 1.0f`
`floatToInt(p, BS)`:
`out.x = (p.X + (p.X > 0 ? BS / 2 : -BS / 2)) / BS`

Both first calculate as f32, and then round. The results should be (apart from floating point inaccuracies, which are acceptable here) both the same.
I don't see why to round here already.

Reply:
i don't understand why you are not directly dividing. Maybe compiler will optimize but it's not mathematically short